File: Examples/CSharp/AsposePDF/Text/ExtractTextAll.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;
using Aspose.Pdf.Text;

namespace CSharp.AsposePdf.Text
{
    public class ExtractTextAll
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdf_Text();

            //open document
            Document pdfDocument = new Document(dataDir + "ExtractTextAll.pdf");

            //create TextAbsorber object to extract text
            TextAbsorber textAbsorber = new TextAbsorber();

            //accept the absorber for all the pages
            pdfDocument.Pages.Accept(textAbsorber);

            //get the extracted text
            string extractedText = textAbsorber.Text;

            // create a writer and open the file
            TextWriter tw = new StreamWriter(dataDir + "extracted-text.txt");

            // write a line of text to the file
            tw.WriteLine(extractedText);

            // close the stream
            tw.Close();
            
            
        }
    }
}

File: Examples/VisualBasic/AsposePDF/Text/ExtractTextAll.vb
Imports System.IO
Imports System
Imports Aspose.Pdf
Imports Aspose.Pdf.Text

Namespace AsposePDF.Text
    Public Class ExtractTextAll
        Public Shared Sub Run()
            ' ExStart:ExtractTextAll
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdf_Text()

            ' Open document
            Dim pdfDocument As New Document(dataDir & Convert.ToString("ExtractTextAll.pdf"))

            ' Create TextAbsorber object to extract text
            Dim textAbsorber As New TextAbsorber()
            ' Accept the absorber for all the pages
            pdfDocument.Pages.Accept(textAbsorber)
            ' Get the extracted text
            Dim extractedText As String = textAbsorber.Text
            ' Create a writer and open the file
            Dim tw As TextWriter = New StreamWriter(dataDir & Convert.ToString("extracted-text.txt"))
            ' Write a line of text to the file
            tw.WriteLine(extractedText)
            ' Close the stream
            tw.Close()
            ' ExEnd:ExtractTextAll   

        End Sub
    End Class
End Namespace
